### PR TITLE
remove part from storage client interface

### PR DESCRIPTION
**Pull request by critical27**
TODO: 
1. batch put/get/remove still have bugs.
2. PartId may be different from c++ implement, need to fix, I have not find a good solution...

### CODE REVIEW

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClient.java [hunk 1]
```diff
@@ -14,19 +14,17 @@
...
+    public boolean put(int space, String key, String value);
 
-    public boolean put(int part, String key, String value);
+    public boolean put(int space, Map<String, String> kvs);
 
-    public boolean put(int part, Map<String, String> values);
+    public Optional<String> get(int space, String key);
 
-    public Optional<String> get(int part, String key);
+    public Optional<Map<String, String>> get(int space, List<String> keys);
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/storage/client/StorageClient.java]: Nice catch !

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java [hunk 2]
```diff
@@ -468,6 +574,38 @@ public boolean removeRange(int part, String start, String end) {
         }
         return false;
     }
+     */
+
+    private boolean doRemove(int space, StorageService.Client client, RemoveRequest request) {
+        ExecResponse response;
+        int retry = connectionRetry;
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: executor retry?

Comment (critical27) [reply client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: You mean retry in excutor not here?

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java [hunk 3]
```diff
@@ -191,57 +139,86 @@ public boolean put(int part, String key, String value) {
...
+                    request.parts.put(part, entry.getValue());
+                } else {
+                    request.parts.get(part).addAll(entry.getValue());
+                }
+            }
+        }
+
+        final CountDownLatch countDownLatch = new CountDownLatch(groups.size());
+        final List<Boolean> responses = Collections.synchronizedList(new ArrayList<Boolean>(groups.size()));
+        ExecutorService executorService = Executors.newFixedThreadPool(groups.size());
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: why don't move `executorService` into class instance scope ? `executorService` maybe a heavily object, create and release frequently would influence performance.

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java [hunk 4]
```diff
@@ -276,71 +250,116 @@ public boolean put(int part, Map<String, String> values) {
...
+        }
+        try {
+            countDownLatch.await();
+        } catch (InterruptedException e) {
+            LOGGER.error("Put interrupted");
+            return Optional.empty();
+        }
+        executorService.shutdown();
+
+        // TODO: judge if we get all values
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: When the code run into this position, `CountDownLatch` have release.

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java [hunk 5]
```diff
@@ -276,71 +250,116 @@ public boolean put(int part, Map<String, String> values) {
...
     @Override
-    public Optional<Map<String, String>> get(int part, List<String> keys) {
-        checkLeader(part);
-        GetRequest request = new GetRequest();
-        Map<Integer, List<String>> parts = Maps.newHashMap();
-        parts.put(part, keys);
-        request.setSpace_id(space);
-        request.setParts(parts);
-        LOGGER.debug(String.format("Get Request: %s", request.toString()));
+    public Optional<Map<String, String>> get(int space, List<String> keys) {
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: Do we know space ID ? Maybe we need something to transform SpaceName into Space ID

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java (hunk 3 above)
Comment (critical27) [reply client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: :thumb:

File: client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java (hunk 5 above)
Comment (critical27) [reply client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: Ageed, we need to support in meta client later

Comment (darionyaphet) [reply client/src/main/java/com/vesoft/nebula/storage/client/StorageClientImpl.java]: ^_^

**Comment by critical27** [comment]
Ready to go, batch put/get/remove still have bugs, maybe fix in another pr.

File: client/src/main/java/com/vesoft/nebula/meta/client/MetaClientImpl.java [hunk 6]
```diff
@@ -188,7 +188,7 @@ public Integer getEdgeType(String spaceName, String edgeName) {
         return getEdgeType(spaceNames.get(spaceName), edgeName);
     }
 
-    private void init() {
+    public void init() {
```
Comment (wfystx) [inline client/src/main/java/com/vesoft/nebula/meta/client/MetaClientImpl.java]: I think there's no need to use init() cause it was already called in constructor

Comment (darionyaphet) [reply client/src/main/java/com/vesoft/nebula/meta/client/MetaClientImpl.java]: We setting member variable in `Constructor`, another operation move into a method is better.

**Pull request merged**